src/common/utxobased/keymanager/keymanager.ts

```diff
@@ -493,100 +555,133 @@ export function derivationLevelScriptHash(): number {
 }
 
 export function xpubToPubkey(args: XPubToPubkeyArgs): string {
-  const network: BitcoinJSNetwork = bip32NetworkFromCoin({
-    networkType: args.network,
-    coinString: args.coin,
-    sigType: args.type
-  })
-  const coin = getCoinFromString(args.coin)
-  const bip32FromBase58Func = coin.bip32FromBase58Func ?? bip32.fromBase58
-  const node: bip32.BIP32Interface = bip32FromBase58Func(args.xpub, network)
-  return node
-    .derive(args.bip44ChangeIndex)
-    .derive(args.bip44AddressIndex)
-    .publicKey.toString('hex')
+  let prefixIndex = 0
+  while (true) {
+    try {
+      const network: BitcoinJSNetwork = bip32NetworkFromCoin({
+        networkType: args.network,
+        coinString: args.coin,
+        sigType: args.type,
+        prefixIndex
+      })
+      const coin = getCoinFromString(args.coin)
+      const bip32FromBase58Func = coin.bip32FromBase58Func ?? bip32.fromBase58
+      const node: bip32.BIP32Interface = bip32FromBase58Func(args.xpub, network)
+      return node
+        .derive(args.bip44ChangeIndex)
+        .derive(args.bip44AddressIndex)
+        .publicKey.toString('hex')
+    } catch (err) {
+      if (isError(err)) {
+        if (err.message === 'Invalid network version') {
+          prefixIndex += 1
+          continue
+        }
+      }
+      throw err
+    }
+  }
```

Comment:
Hmm, this loop pattern doesn't show a clear indication of halting. I'm wary of infinite loops cropping up if this code decays to that overtime. Perhaps a better approach would be to loop over all prefixes and throw outside that loop. This would make it clear that the iterations are bounded by the items in the set of derived prefixes by the `prefixIndex`.